client/client.go

```diff
@@ -1025,11 +1025,20 @@ func (c *Client) retryIntv(base time.Duration) time.Duration {
 // registerAndHeartbeat is a long lived goroutine used to register the client
 // and then start heartbeatng to the server.
 func (c *Client) registerAndHeartbeat() {
+	// Before registering capture the hashes of the Node. The hashes may be out
+	// of date with what registers but this is okay since the loop checking for
```

Comment:
"with what registers" - if this is for Consul specifically, add that detail.

Reply:
This is for registering against the Nomad Server. With the `retryRegisterNode` call below I think there is context